### ISSUE TITLE
issues found in 'supdeq_shiftDistance' function

### ISSUE THREAD

**Issue by YLiAcoustics** [AudioGroupCologne/SUpDEq#4]
Dear authors of the SUPDEQ toolbox,

First of all, thanks a lot for providing the toolbox!

I am using 'supdeq_shiftDistance.m' for computing near-field HRTFs from far-field HRTFs. As far as I understand, it is the implementation of the DVF method by Kan et al. (2009).

During implementation, I found 2 issues in the code:

**_line 43: sg=[targetDirection(1),90-targetDirection(2),startingDistance;targetDirection(1),90-targetDirection(2),targetDistance];_**
The input 'targetDirection' is a Qx2 matrix, therefore I assume
sg=[targetDirection(:,1),90-targetDirection(:,2),startingDistance;targetDirection(:,1),90-targetDirection(:,2),targetDistance];
would be correct, given Q>1.

**_line 48: hDistances=AKsphericalHead(sg,ear,false,headRadius,sg(1,3),100,length(startingHRTF_L)*2-2,fs,c);_**
where sg(1,3) = StartingDistance is used as input 'r_0' for function 'AKsphericalHead(sg, ear, offCenter, a, r_0, Nsh, Nsamples, fs, c)'. As r_0 corresponds to the distance from the source to the center of the spherical head, it seems to me that this line only calculates the spherical head impulse response at the starting position. This is also because the input 'sg' for AKsphericalHead does not work for various distances in the 3rd column.
The way I am implementing this function is as follows:
_hstarting=AKsphericalHead(startingsg,ear,false,headRadius,startingDistance,100,length(startingHRTF_L)*2-2,fs,c);                                                                                         
htarget=AKsphericalHead(targetsg,ear,false,headRadius,targetDistance,100,length(startingHRTF_L)*2-2,fs,c);_
 
I am looking forward to your reply.

Yuqing.

**Comment by JMA-JMA** [comment]
Dear Yuqing,

first of all, thank you for using the SUpDEq toolbox and thanks for your request. Sorry for the late response, though. Noone is really monitoring this Git's issue section. I will have an eye on this from now on...

The `supdeq_shiftDistance` function was solely implemented for distance error compensation (see function `supdeq_dec`) as applied when measuring HRTFs or voice directivity to compensate for position inaccuracies. Please see the following publications concerning the implementation:

> Pörschmann, C., & Arend, J. M. (2019). How positioning inaccuracies influence the spatial upsampling of sparse head-related transfer function sets. Proceedings of the International Conference on Spatial Audio (ICSA), Ilmenau, Germany, 47–54. https://doi.org/10.22032/dbt.39952
> 
> Pörschmann, C., & Arend, J. M. (2020). A Method for Spatial Upsampling of Voice Directivity by Directional Equalization. J. Audio Eng. Soc., 68(9), 649–663. https://doi.org/10.17743/jaes.2020.0033

As it uses spherical head / rigid sphere functions for distance compensation, it can generally also be used for distance shifts of HRTFs similar to the DVF method from Kan et al. (2009). 

However, with the function `supdeq_dvf`, we implemented a much more comprehensive and easier-to-use function for distance shifts of HRTFs (and near-field HRTF synthesis based on far-field datasets) using Kan's DVF method in combination with acoustic parallax processing. The function was used to synthesize near-field HRTFs for the following publication:

> Arend, J. M., Ramírez, M., Liesefeld, H. R., & Pörschmann, C. (2021). Do near-field cues enhance the plausibility of non-individual binaural rendering in a dynamic multimodal virtual acoustic scene? Acta Acustica, 5(55), 1–14. https://doi.org/10.1051/aacus/2021048

I would highly recommend to use this function for near-field HRTF synthesis when using DVFs. As an alternative method, the function `supdeq_rangeExt` allows shifting HRTFs in SH domain using radial functions.

That said, I will go through your issue comment and check if the `supdeq_shiftDistance` function needs to be adjusted. As we want to keep the code functioning for its original purpose (distance shifts for correcting position inaccuracies), I am however not sure on how much I can change. I would, as described above, suggest to generally use the `supdeq_dvf` function instead.

Best,
Johannes

**Comment by JMA-JMA** [comment]
Dear Yuqing,

We reviewed the code but decided to keep the function `supdeq_shiftDistance` as it is, so that previous implementations run without further issues. As described above and in the README, the function `supdeq_dvf` should be used for near-field HRTF synthesis using the DVF-method. 

We apologize for the confusion. To stay more consistent, we also deleted the `supdeq_demo_DVF` script, as it was an older version of the DVF implementation working only in SH domain in combination with spatial upsampling of the HRTF dataset, which is not necessarily required. 

If interested, the supplementary material to the article
> Arend, J. M., Ramírez, M., Liesefeld, H. R., & Pörschmann, C. (2021). Do near-field cues enhance the plausibility of non-individual binaural rendering in a dynamic multimodal virtual acoustic scene? Acta Acustica, 5(55), 1–14. https://doi.org/10.1051/aacus/2021048

which can be found in this Zenodo repository https://doi.org/10.5281/zenodo.5656726
provides Matlab material using the `supdeq_dvf` function for near-field HRTF synthesis.

Best,
Johannes
